modules/ding_adgangsplatformen/ding_adgangsplatformen.module

```diff
@@ -275,20 +275,50 @@ function ding_adgangsplatformen_login() {
 function ding_adgangsplatformen_logout($regen_session = FALSE) {
   global $base_url;
 
+  // Build base options for logging out.
   $config = ding_adgangsplatformen_get_configuration();
   $singleLogout = $config['singleLogout'] ? 'true' : 'false';
   $token = ding_adgangsplatformen_openplatform_token_get();
 
+  $logout_options = [
+    'singlelogout' => $singleLogout,
+    'redirect_uri' => $base_url,
+    'token' => $token,
+  ];
+
+  if (ding_adgangsplatformen_has_token_client()) {
+    // Revoke our current token. It has been retrieved by the token provider using
+    // password grants and will not be revoked by Adgangsplatformen itself during
+    // logout.
+    ding_adgangsplatformen_revoke_token(
+      ding_adgangsplatformen_get_provider(),
+      $token
+    );
+
+    try {
+      // Retrieve a new anonymous token and use that for the logout procedure. By
+      // using the authentication provider we ensure that Adgangsplatformen will
+      // redirect to the right site.
+      $auth_provider = ding_adgangsplatformen_get_provider(TRUE);
+      $token = $auth_provider->getAccessToken('password', [
+        // Using @ as username and password works as an anonymous user.
+        'username' => '@',
+        'password' => '@',
+      ]);
```

Comment:
I not sure here, but from usage of the open platform... should the `ting_agency` have been sent with the request here?

Reply:
In the form `@775100` for both username and password? It's how we do it af the open platform at authentication to get an anonymous token for a given library?

Reply:
Not as I understand the situation. According to @vibjerg the important part here is the client id as that it is the client that knows which urls are allowed for logout redirects.